src/test/java/seedu/address/model/entry/EntryNameTest.java

```diff
@@ -0,0 +1,43 @@
+package seedu.address.model.entry;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.testutil.Assert.assertThrows;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.model.person.Name;
+
+public class EntryNameTest {
+
+    @Test
+    public void constructor_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new EntryName(null));
+    }
+
+    @Test
+    public void constructor_invalidEntryName_throwsIllegalArgumentException() {
+        String invalidName = "";
+        assertThrows(IllegalArgumentException.class, () -> new EntryName(invalidName));
+    }
+
+    @Test
+    public void isValidName() {
+        // null name
+        assertThrows(NullPointerException.class, () -> EntryName.isValidName(null));
+
+        // invalid name
+        assertFalse(EntryName.isValidName(""));
+        assertFalse(EntryName.isValidName(" "));
+        assertFalse(EntryName.isValidName("@"));
+        assertFalse(EntryName.isValidName("hell@"));
+        assertFalse(EntryName.isValidName("jon        *"));
+
+        // valid name
+        assertTrue(Name.isValidName("peter jack")); // alphabets only
```

Comment:
The names given for these tests should sound more like entry names and not contact names, same for the assertFalse statements above.